### PR TITLE
ch.bfe.minergiegebaeude

### PR DESCRIPTION
**Pull request by rebert**
[try](https://mf-geoadmin3.dev.bgdi.ch/?api_url=%2F%2Fmf-chsdi3.dev.bgdi.ch%2Fltmae_minergie&topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege,ch.bfe.minergiegebaeude&layers_visibility=false,false,false,false,true&layers_timestamp=18641231,,,,)

### CODE REVIEW

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<%inherit file="base.mako"/>
+
+<%def name="table_body(c, lang)">
+<%
+    if lang == 'de':
```
Comment (AFoletti) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: What about RM?
You could use DE as fallback. Something like:
if FR
[*_fr]
elif EN
[*_en]
elif IT
[*_it]
else
[*_de]

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        http = c['attributes']['http_fr']
+    elif lang == 'it':
+        info == c['attributes']['buildinginfo_it']
+        http = c['attributes']['http_it']
+    elif lang == 'en':
+        info == c['attributes']['buildinginfo_en']
+        http = c['attributes']['http_en']
+%>
+    <% c['stable_id'] = True %>
+    <tr><td class="cell-left-large">${_('ch.bfe.minergiegebaeude.zertifikat')}</td><td>${c['attributes']['certificate'] or '-'}</td></tr>
```
Comment (AFoletti) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: wrong CSS class... left cell is too big with this one

**Comment by AFoletti** [comment]
Thanks! Looks good but I've a couple comments in the code.
Moreover:
- your WMS gutter looks too small
image
- remove default transparency
- legends are missing

File: chsdi/models/vector/uvek.py [hunk 3]
```diff
@@ -604,6 +604,35 @@ class ENERGIEFORSCHUNG(Base, Vector):
...
+    __bodId__ = 'ch.bfe.minergiegebaeude'
+    __extended_info__ = False
+    __label__ = 'id'
+    id = Column('bgdi_id', Integer, primary_key=True)
+    certificate = Column('certificate', Text)
+    standard = Column('standard', Text)
+    street = Column('street', Text)
+    streetnr = Column('streetnr', Text)
+    postcode = Column('postcode', Numeric)
+    place = Column('place', Text)
```
Comment (loicgasser) [inline chsdi/models/vector/uvek.py]: Unicode

File: chsdi/models/vector/uvek.py [hunk 4]
```diff
@@ -604,6 +604,35 @@ class ENERGIEFORSCHUNG(Base, Vector):
 register('ch.bfe.energieforschung', ENERGIEFORSCHUNG)
 
 
+class MinergieGebaeude(Base, Vector):
+    __tablename__ = 'minergiegebaeude'
+    __table_args__ = ({'schema': 'bfe', 'autoload': False})
+    __template__ = 'templates/htmlpopup/minergiegebaeude.mako'
+    __bodId__ = 'ch.bfe.minergiegebaeude'
+    __extended_info__ = False
```
Comment (loicgasser) [inline chsdi/models/vector/uvek.py]: If you don't put the attribute -> means no extended info, so you could remove it.

**Comment by AFoletti** [comment]
thanks @rebert. Please deploy ASAP to int so that I can send the link to the customer

**Comment by oterral** [comment]
jenkins test

**Comment by rebert** [comment]
@loicgasser: Being deployed tomorrow

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        http = c['attributes']['http_fr']
+    elif lang == 'it':
+        info = c['attributes']['buildinginfo_it']
+        http = c['attributes']['http_it']
+    elif lang == 'en':
+        info = c['attributes']['buildinginfo_en']
+        http = c['attributes']['http_en']
+    else:
+        info = c['attributes']['buildinginfo_de']
+        http = c['attributes']['http_de']
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: simply do

```python
lang = 'de' if lang == 'rm' else lang
info = c['attributes']['buildinginfo_%s' % lang]
http = c['attributes']['http_%s' % lang]
````

File: chsdi/models/vector/uvek.py [hunk 6]
```diff
@@ -604,6 +604,34 @@ class ENERGIEFORSCHUNG(Base, Vector):
...
+    street = Column('street', Unicode)
+    streetnr = Column('streetnr', Unicode)
+    postcode = Column('postcode', Integer)
+    place = Column('place', Unicode)
+    canton = Column('canton', Unicode)
+    buildinginfo_de = Column('buildinginfo_de', Unicode)
+    buildinginfo_fr = Column('buildinginfo_fr', Unicode)
+    buildinginfo_it = Column('buildinginfo_it', Unicode)
+    buildinginfo_en = Column('buildinginfo_en', Unicode)
+    ebf = Column('ebf', Numeric)
```
Comment (loicgasser) [inline chsdi/models/vector/uvek.py]: no Numeric

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+%>
+    <% c['stable_id'] = True %>
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: bgdi_id not unique, you can change id if they provide any

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.zertifikat')}</td><td>${c['attributes']['certificate'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.standard')}</td><td>${c['attributes']['standard'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.adresse')}</td><td>${c['attributes']['street'] or '-'}&nbsp;${c['attributes']['streetnr'] or '-'}</td></tr>
+    <tr><td class="cell-left">&nbsp;</td><td>${c['attributes']['postcode'] or '-'}&nbsp;${c['attributes']['place'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.kanton')}</td><td>${c['attributes']['canton'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.gebaeude')}&nbsp;(${_('ch.bfe.minergiegebaeude.energiebezugsflaeche')}&nbsp;m<sup>2</sup>)</td><td>${info}&nbsp;(${c['attributes']['ebf'] or '-'}&nbsp;m<sup>2</sup>)</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.weitere_informationen')}</td><td><a href="${http}" target="_blank">${_('ch.bfe.minergiegebaeude')}</a></td></tr>
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: % if http is not None:
<tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.weitere_informationen')}</td><td><a href="${http}" target="_blank">${_('ch.bfe.minergiegebaeude')}</a></td></tr>
% endif

File: chsdi/templates/htmlpopup/minergiegebaeude.mako [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        http = c['attributes']['http_de']
+
+%>
+    <% c['stable_id'] = True %>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.zertifikat')}</td><td>${c['attributes']['certificate'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.standard')}</td><td>${c['attributes']['standard'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.adresse')}</td><td>${c['attributes']['street'] or '-'}&nbsp;${c['attributes']['streetnr'] or '-'}</td></tr>
+    <tr><td class="cell-left">&nbsp;</td><td>${c['attributes']['postcode'] or '-'}&nbsp;${c['attributes']['place'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.kanton')}</td><td>${c['attributes']['canton'] or '-'}</td></tr>
+    <tr><td class="cell-left">${_('ch.bfe.minergiegebaeude.gebaeude')}&nbsp;(${_('ch.bfe.minergiegebaeude.energiebezugsflaeche')}&nbsp;m<sup>2</sup>)</td><td>${info}&nbsp;(${c['attributes']['ebf'] or '-'}&nbsp;m<sup>2</sup>)</td></tr>
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/minergiegebaeude.mako]: ${info or `-`}

**Comment by rebert** [comment]
@loicgasser Thanks for your feedback, which has been implemented

**Pull request merged**